python/cudf/cudf/core/column/numerical.py

```diff
@@ -254,7 +254,7 @@ def default_na_value(self):
         if dkind == "f":
             return self.dtype.type(np.nan)
         elif dkind in {"u", "i"}:
-            return self.dtype.type(0)
+            return self.dtype.type(np.iinfo("int64").max)
```

Comment:
This doesn't seem safe. What if the dtype is int8 or int16 or uint32?

Reply:
It will appropriately change to `-1` or `255` in case of `int8` or `uint8`.

Reply:
I think we should just more explicitly use `min(dtype)` for signed integers and `max(dtype)` for unsigned integers. Using `max(int64)` for `uint64` feels gross

Reply:
Changed the sentinel value to be max(int64) rather than max(uint64) as the value of an int can't be greater than sys.maxsize, atleast in case `uint64`.

Reply:
That might work, will take a look.